frontend/hub/access/roles/RolePage/HubRoleForm.tsx

```diff
@@ -0,0 +1,177 @@
+import { useTranslation } from 'react-i18next';
+import { useNavigate, useParams } from 'react-router-dom';
+import { useWatch } from 'react-hook-form';
+import {
+  PageFormSelect,
+  PageFormSubmitHandler,
+  PageFormTextInput,
+  PageHeader,
+  PageLayout,
+  useGetPageUrl,
+  usePageNavigate,
+} from '../../../../../framework';
+import { useInvalidateCacheOnUnmount } from '../../../../common/useInvalidateCache';
+import { usePostRequest } from '../../../../common/crud/usePostRequest';
+import { HubRbacRole } from '../../../interfaces/expanded/HubRbacRole';
+import { hubAPI } from '../../../common/api/formatPath';
+import { HubRoute } from '../../../main/HubRoutes';
+import { HubPageForm } from '../../../common/HubPageForm';
+import { useHubRoleMetadata } from '../hooks/useHubRoleMetadata';
+import { PageFormHidden } from '../../../../../framework/PageForm/Utils/PageFormHidden';
+import { PageFormMultiSelect } from '../../../../../framework/PageForm/Inputs/PageFormMultiSelect';
+import { ContentTypeEnum } from '../../../interfaces/expanded/ContentType';
+import { useGet } from '../../../../common/crud/useGet';
+import { usePatchRequest } from '../../../../common/crud/usePatchRequest';
+import { useIsValidRoleName } from '../hooks/useIsValidRoleName';
+
+export function CreateRole(props: { breadcrumbLabelForPreviousPage?: string }) {
+  const { t } = useTranslation();
+  const navigate = useNavigate();
+  const pageNavigate = usePageNavigate();
+
+  useInvalidateCacheOnUnmount();
```

Comment:
Do we need this one?

Reply:
Honestly, I'm not sure. I see that both Keith and Mike used it in their implementations of the AWX and EDA roles. @keithjgrant do you know why this is needed?

Reply:
I copied that from the EDA form, so I'm not 100% sure. My guess is it ensures that SWR reloads a fresh copy of the Role object after it redirects to the details page when saving.

Reply:
Thanks @keithjgrant ! @nixocio I'm inclined to keep this as it's been working on the other roles UIs.